api/webhook/post.go

```diff
@@ -664,6 +666,53 @@ func PostWebhook(c *gin.Context) {
 	// set the BuildID field
 	h.SetBuildID(b.GetID())
 
+	// if event is deployment, update the deployment record to include this build
+	if b.GetEvent() == constants.EventDeploy {
+		builds := []*library.Build{}
+		builds = append(builds, b)
+
+		d, err := database.FromContext(c).GetDeploymentForRepo(c, repo, webhook.Deployment.GetNumber())
+		if err != nil {
+			if errors.Is(err, gorm.ErrRecordNotFound) {
+				deployment := webhook.Deployment
+
+				deployment.SetRepoID(repo.GetID())
+				deployment.SetBuilds(builds)
+
+				_, err := database.FromContext(c).CreateDeployment(c, deployment)
+				if err != nil {
+					retErr := fmt.Errorf("%s: failed to create deployment %s/%d: %w", baseErr, repo.GetFullName(), deployment.GetNumber(), err)
+					util.HandleError(c, http.StatusInternalServerError, retErr)
+
+					h.SetStatus(constants.StatusFailure)
+					h.SetError(retErr.Error())
+
+					return
+				}
+			} else {
+				retErr := fmt.Errorf("%s: failed to get deployment %s/%d: %w", baseErr, repo.GetFullName(), webhook.Deployment.GetNumber(), err)
+				util.HandleError(c, http.StatusInternalServerError, retErr)
+
+				h.SetStatus(constants.StatusFailure)
+				h.SetError(retErr.Error())
+
+				return
+			}
+		} else {
+			d.SetBuilds(builds)
```

Comment:
```suggestion
			d.SetBuilds([]*library.Build{b})
```

If you do it this way you can remove the lines above where you set `builds`.